src/core/net/ip6.cpp

```diff
@@ -716,7 +717,7 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, int8_t interface
         {
             forward = true;
 
-            if (fromLocalHost)
+            if (fromLocalHost && header.GetDestination().GetScope() >= Address::kRealmLocalScope)
```

Comment:
nit: `InsertMplOption()` did not return error when the scope was link-local to avoid having this check twice.